lib/libc/minimal/source/stdout/stdout_console.c

```diff
@@ -40,7 +41,18 @@ static inline int z_vrfy_zephyr_fputc(int c, FILE *stream)
 
 int fputc(int c, FILE *stream)
 {
-	return zephyr_fputc(c, stream);
+	int rc;
+	struct k_mutex *lock = z_get_fd_lock(POINTER_TO_INT(stream) - 1);
```

Comment:
Could we check that the c and stream are somewhat sane values? Also need to update errno for failures.

Reply:
I agree - rather than exposing the mutex, I think it would make more sense to mimic what is done with `flockfile()` / `funlockfile()` but with integer file descriptors rather than `FILE *`.

Maybe `zvfs_lockfile()` / `zvfs_unlockfile()`?

Also, `zvfs_trylockfile()` for completeness